bundle-workflow/python/build_workflow/builder.py

```diff
@@ -0,0 +1,48 @@
+# Copyright OpenSearch Contributors.
```

Comment:
This looks like a variation/reimplementation of https://github.com/opensearch-project/opensearch-build/blob/main/bundle-workflow/python/lib/component.py without dataclass.